src/pages/ThorChainLP/queries/hooks/useSendThorTx.tsx

```diff
@@ -0,0 +1,458 @@
+                // https://dev.thorchain.org/concepts/sending-transactions.html#admonition-info-1
+                zeroAddress,
+          amount: amountOrDustCryptoBaseUnit,
+          memo,
+          expiry: BigInt(dayjs().add(15, 'minute').unix()),
+        })
+
+        return {
+          // amountCryptoPrecision is always denominated in fee asset - the only value we can send when calling a contract is native asset value
+          // which happens for deposits (0-value) and withdrawals (dust-value, failure to send it means Txs won't be seen by THOR)
+          amountCryptoPrecision:
+            // TODO(gomes): this may not be applicable to other domains - verify the validity of this for others and adapt accordingly
+            isToken(fromAssetId(assetId).assetReference) && thorfiAction === 'ThorchainLpEnter'
+              ? '0'
+              : fromBaseUnit(amountOrDustCryptoBaseUnit, feeAsset.precision),
+          // Withdrawals do NOT occur a dust send to the contract address.
+          // It's a regular 0-value contract-call
+          // TODO(gomes): double check that this logic is correct across all domains, it is critical and getting things wrong here can lead to funds being lost
+          assetId: thorfiAction === 'ThorchainLpExit' ? feeAsset.assetId : asset.assetId,
+          to: inboundAddressData.router,
+          from: assetAddress,
+          sendMax: false,
+          // This is an ERC-20, we abuse the memo field for the actual hex-encoded calldata
+          memo: data,
+          accountId,
+          // Note, this is NOT a send.
+          // contractAddress is only needed when doing a send and the account interacts *directly* with the token's contract address.
+          // Here, the LP contract is approved beforehand to spend the token value, which it will when calling depositWithExpiry()
+          contractAddress: undefined,
+        }
+      }
+      case 'Send': {
+        if (!inboundAddressData || !assetAddress) return undefined
+        return {
+          // TODO(gomes): When implementing this for savers, we will want to ensure that dust amount is only sent for non-UTXO chains
+          // EVM chains should make use of depositWithExpiry() for withdrawals
+          amountCryptoPrecision:
+            thorfiAction === 'ThorchainLpExit' ? dustAmountCryptoPrecision : amountCryptoPrecision,
+          assetId,
+          to: inboundAddressData.address,
+          from: assetAddress,
+          sendMax: false,
+          memo,
+          accountId,
+          contractAddress: undefined,
+        }
+      }
+      default:
+        return undefined
+    }
+  }, [
+    accountId,
+    amountCryptoPrecision,
+    asset,
+    assetAddress,
+    assetId,
+    feeAsset,
+    inboundAddressData,
+    memo,
+    shouldUseDustAmount,
+    thorfiAction,
+    transactionType,
+    wallet,
+  ])
+
+  const { data: estimatedFeesData } = useGetEstimatedFeesQuery({
```

Comment:
along with this hook, we should move the `useGetEstimatedFeesQuery` hook to a common thorchain utils location.

todo: decouple this hook from the send modal utils and leverage `lib/utils/evm.ts` so we can honor the correct eip1559. it may be beneficial to follow the evm utils pattern for the other chains to standardize the api across chains. this would also allow us to unify `estimate` vs `onSignTx` functions in this hook. they are basically doing the same thing regarding estimation/build

Reply:
Next steps here:

1. instead of consuming `estimateFees()` in `useGetEstimatedFeesQuery`, use `getFees()` `getFeesWithWallet` which is *not* hacked around sends, the current `estimateFees` abusing the `memo` field for calldata (and also missing EIP1559 fees estimation)

In other words, all 3 branches should have their fees estimated with the same `CreateBuildCustomTxInputArgs` payload.

To get there, we will most likely want to decouple things into: `createTxInput`, which can be used both for estimating fees and for building/broadcasting. This already exists for EVM as `createBuildCustomTxInput`, so we'll need to build something similar for other chains